### PR TITLE
add product detail page

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/76016

## WHAT (optional)
- [Guest] View product detail

## HOW
- Add product detail page for guest

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controllers/product.controller.ts [hunk 1]
```diff
@@ -17,4 +19,29 @@ export class ProductController {
...
+
+  public getProductDetail = asyncHandler(
+    async (req: Request, res: Response) => {
+      const product = await this.getProductFromIdParam(req, res);
+      if(product === null) {
+        req.flash('error', Error.PRODUCT_NOT_FOUND);
+        res.redirect('/');
+      }
+      res.render('user/product/detail', {product});
+      return;
```
Comment (huongnt-2545) [inline src/controllers/product.controller.ts]:
```suggestion
      return res.render('user/product/detail', {product});
```

File: src/controllers/product.controller.ts [hunk 2]
```diff
@@ -17,4 +19,29 @@ export class ProductController {
...
+    },
+  );
+
+  public getProductFromIdParam = async (
```
Comment (huongnt-2545) [inline src/controllers/product.controller.ts]: private method

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**